Bento/Bento/If.swift

```diff
@@ -1,99 +1,69 @@
-public struct If<T> {
-    let condition: () -> Bool
-    let generator: () -> T
-
-    public static func iff(_ condition: @autoclosure @escaping () -> Bool, _ generator: @autoclosure @escaping () -> T) -> If<T> {
-        return If(condition: condition, generator: generator)
-    }
-
-    public static func iff(_ condition: @autoclosure @escaping () -> Bool, _ generator: @escaping () -> T) -> If<T> {
-        return If(condition: condition, generator: generator)
+extension Optional {
+    public static func iff(_ condition: @autoclosure @escaping () -> Bool, _ generator: @autoclosure @escaping () -> Wrapped) -> Wrapped? {
```

Comment:
@inamiy I don't think `@escaping` is really needed, can we remove it? Because of this we have to use `self` in each `.iff`/`.some` call =/

Reply:
Good catch 👍  Fixed in 437bd22 .